pkg/util/clusterdata/worker_profiles_task.go

```diff
@@ -70,7 +70,10 @@ func (ef *workerProfilesEnricherTask) FetchData(callbacks chan<- func(), errs ch
 	workerProfiles := make([]api.WorkerProfile, len(machinesets.Items))
 	for i, machineset := range machinesets.Items {
 		if machineset.Spec.Template.Spec.ProviderSpec.Value == nil {
-			continue
```

Comment:
It looks like nil pointer dereference itself got fixed as part of #764. But I think we should not continue here: it will be super confusing to see 2 machinesets when you actually have 3 (one malformed).

Reply:
+1, the commit in #764 was a mistake on my part.

Reply:
Hmm, thinking about it, would it be best in this case to emit a WorkerProfile with machineProviderSpec-related fields empty?

Reply:
`WorkerProfile` consists mostly from `machineProviderSpec` fields. We will only show name and count of VMs. But if machineset is malformed - it might mean that the count is wrong. Also confusing. I think it's better to not show `WorkerProfile`s here at all.

Reply:
I think so too. Less confusion

Reply:
Discussed in Slack. Posting for visibility:

* We want to show the desired state. Even if the desired state is unachievable (such as machine set has number of replicas specified, but no provider spec)
* Showing something to the customer is better than hiding all the machine sets when the problem is only with one

Reply:
@jim-minter I updated the function and relevant test. Please take another look.